Composer/packages/lib/ui-shared/src/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as React from 'react';
+
+/**
+ * Creates a textarea at then end of the document body then
+ * uses that to copy the specified text to the cliboard.
+ */
+
+export const copyToClipboard = (text: string) => {
+  // Remember the current selection
+  const rangeCount = document?.getSelection()?.rangeCount || 0;
+  const selected = rangeCount > 0 ? document?.getSelection()?.getRangeAt(0) : false;
+
+  // Create an offscreen textarea and copy the text
+  const element = document.createElement('textarea');
+  element.value = text;
+  element.setAttribute('readonly', '');
+  element.style.position = 'absolute';
+  element.style.left = '-9999px';
+  document.body.appendChild(element);
+  element.select();
+  const success = document.execCommand('copy');
+  document.body.removeChild(element);
+
+  // Restore the previous selection
+  if (document?.getSelection() && selected) {
```

Comment:
why the document?.getSelection() check?

Reply:
Was trying to avoid the ?. below, but they end up being needed.